src/queries/function_parameter_count_changed.ron

```diff
@@ -0,0 +1,50 @@
+SemverQuery(
+    id: "function_parameter_count_changed",
+    human_readable_name: "pub fn parameter count changed",
+    description: "Parameter count of a function has changed.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#fn-change-arity"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output @tag
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        old_parameter_: parameter @fold @transform(op: "count") @output @tag(name: "parameters")
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @filter(op: "=", value: ["%name"])
+
+                        current_parameter_: parameter @fold @transform(op: "count") @filter(op: "!=", value: ["%parameters"]) @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+    },
+    error_message: "Parameter count of a publicly-visible function has changed.",
+    per_result_error_template: Some("parameter count of function {{join \"::\" path}} changed from {{old_parameter_count}} to {{current_parameter_count}}, in {{span_filename}}:{{span_begin_line}}"),
```

Comment:
Slight wording tweak suggestions for these user-facing messages — see the other query below, they are analogous.